service-base/src/main/java/org/eclipse/hono/service/tenant/TenantService.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.service.tenant;
+
+import java.net.HttpURLConnection;
+
+import org.eclipse.hono.util.TenantResult;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Verticle;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A service for keeping record of tenant information.
+ *
+ */
+public interface TenantService extends Verticle {
+
+    /**
+     * Gets tenant data by tenant ID.
+     *
+     * @param tenantId The ID of he tenant for which data is requested.
+     * @param resultHandler The handler to invoke with the result of the operation. If tenant exists, the
```

Comment:
IMHO it would be smarter to refer to the API spec wrt the status codes so we do not need to manually keep this JavaDoc in sync. WDYT?

Reply:
I see your point, but since this is **the** interface to implement, when the service is implemented on your own, it is quite helpful IMHO to have the exact definition of the status codes available here in Java (instead of having to click on a link (which by the way is not versioned currently if it is the Eclipse page)).

So I would prefer to have the unconvenience in this case on our side, meaning: we have to manually keep this in sync. Should be very rarely, don't you think so?